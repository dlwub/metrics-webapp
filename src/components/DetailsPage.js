import React, { useState } from 'react';
import { Link } from 'react-router-dom';
import { useSelector, useDispatch } from 'react-redux';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faAngleLeft } from '@fortawesome/free-solid-svg-icons';
import { faArrowAltCircleRight } from '@fortawesome/free-regular-svg-icons';
import Question from './Question';
import { countAnswer, setColor } from '../redux/Questions/Questions';

export default function DetailsPage() {
  const [flag, setFlag] = useState(false);
  const dispatch = useDispatch();
  const count = useSelector((state) => state.questions.count);
  const questions = useSelector((state) => state.questions.list);
  const handleClick = () => {
    setFlag(true);
    dispatch(countAnswer());
    dispatch(setColor());
  };
  const displayQuestions = questions.map((question) => (
    <div key={question.id} className="question-backBtn">
      <Question
        key={question.id}
        id={question.id}
        question={question.question}
        correctAnswer={question.correctAnswer}
        choices={question.choices}
      />
      <Link className="links home-link" to="/">
        <FontAwesomeIcon className="icon-forward" icon={faArrowAltCircleRight} />
      </Link>
    </div>
  ));

  return (
    <>
      <div className="question-bar">
        <Link className="links home-link" to="/">
          <FontAwesomeIcon className="icon" icon={faAngleLeft} />
        </Link>
        <h3 className="questions-title">Questions</h3>
      </div>
      <div className="question-container">{displayQuestions}</div>
      <button type="button" className="check-btn" onClick={() => handleClick()}>Check Answers</button>
      {flag && (
      <div className="view-score">
        <span className="score-span">{`You scored ${count}/5`}</span>
        <Link className="links home-link" to="/">
          <button className="try-btn" type="button">Play Again</button>
        </Link>
      </div>
      )}

    </>
  );
}
