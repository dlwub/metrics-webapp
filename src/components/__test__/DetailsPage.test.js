import React from 'react';
import Renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../../redux/configureStore';
import DetailsPage from '../DetailsPage';

describe('DetailsPage', () => {
  it('renders correctly', () => {
    const tree = Renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <DetailsPage />
        </BrowserRouter>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
