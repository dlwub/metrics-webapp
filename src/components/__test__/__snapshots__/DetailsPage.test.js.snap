// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsPage renders correctly 1`] = `
Array [
  <div
    className="question-bar"
  >
    <a
      className="links home-link"
      href="/"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-angle-left icon"
        data-icon="angle-left"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 320 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M41.4 233.4c-12.5 12.5-12.5 32.8 0 45.3l160 160c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L109.3 256 246.6 118.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-160 160z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </a>
    <h3
      className="questions-title"
    >
      Questions
    </h3>
  </div>,
  <div
    className="question-container"
  />,
  <button
    className="check-btn"
    onClick={[Function]}
    type="button"
  >
    Check Answers
  </button>,
]
`;
